### PR TITLE
 MERC-341 fix state validation in shipping estimator

### PR DESCRIPTION
**Pull request by sherrybc**
- when default country does not have states, show input field 
- update state validation; only validate when it is a dropdown

@mcampa @mickr @bc-ejoe

### CODE REVIEW

File: assets/js/theme/cart/shipping-estimator.js [hunk 1]
```diff
@@ -73,6 +78,34 @@ export default class ShippingEstimator {
         ]);
     }
 
+    setStateValidation() {
+
+        let $last;
+
+        stateCountry(this.$state, this.context, { useIdForStates: true }, (err, field) => {
+            if (err) {
+                throw new Error(err);
```
Comment (mcampa) [inline assets/js/theme/cart/shipping-estimator.js]: Is `err` a string? if not, you should just throw err

**Comment by mickr** [comment]
:+1:

File: templates/components/cart/shipping-estimator.html [hunk 2]
```diff
@@ -34,14 +34,18 @@
...
             <dd class="estimator-form-input">
-                <select class="form-select" id="shipping-state" name="shipping-state" data-field-type="State">
-                    <option>{{lang 'cart.shipping_estimator.select_a_state'}}</option>
-                    {{#each states}}
-                        <option value="{{id}}" {{#if selected}}selected="selected"{{/if}}>
-                            {{name}}
-                        </option>
-                    {{/each}}
-                </select>
+                {{#if selected_state}}
```
Comment (mcampa) [inline templates/components/cart/shipping-estimator.html]: I think you need to check for `states.length` instead of `selected_state`

Comment (mcampa) [inline templates/components/cart/shipping-estimator.html]: Yeah I just check the context. `selected_state` could be null while `states` is an array with all the states

**Comment by mcampa** [comment]
:+1:  LGTM

**Pull request merged**